### PR TITLE
Predifined filter for timestamp

### PR DESCRIPTION
**Pull request by titus6012000**
**MUST BE REVIEWED AFTER THE PULL REQUEST ON UPLOAD FILE HAS BEEN MERGED**

Add a configured predefined filter on the timestamp column in the business date view

### CODE REVIEW

**Review by kahyami — CHANGES_REQUESTED** [review]

File: src/components/filter-menu.js [hunk 1]
```diff
@@ -19,17 +33,63 @@ const FilterMenu = ({ filterHint, handleChange, currentFilter }) => {
...
+
     const handleClose = () => {
         setAnchorElementForFilterMenu(null);
     };
 
     const handleLocalChange = (event) => {
-        setLocalFilter(event.currentTarget.value);
-        handleChange(event);
+        let newtoFilter = [];
+        let boxfilter = [...selectedFilter];
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/filter-menu.js:47]: boxfilter renamed to box**F**ilter

File: src/components/global-view-core.js [hunk 2]
```diff
@@ -136,21 +152,49 @@ const GlobalViewCore = ({ timestampMin, timestampMax, timestampStep }) => {
...
+            setPage(
+                Math.floor(
+                    filterSteps(statusFilter, newFilter).length / rowsPerPage
+                )
+            );
+        }
+    };
+
+    const filterSteps = (currentStatusFilter, currentTimestampFilter) => {
+        // gridcapaFormatDate is not accessible inside the filter we have to use an intermediaire
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/global-view-core.js:183]: wording reviewed

File: src/components/filter-menu.js [hunk 3]
```diff
@@ -19,17 +33,63 @@ const FilterMenu = ({ filterHint, handleChange, currentFilter }) => {
...
+        if (event.currentTarget.name === 'text') {
+            setLocalFilter([event.currentTarget.value]);
+            if (event.currentTarget.value !== '')
+                newtoFilter.push(event.currentTarget.value);
+        } else {
+            let index = event.currentTarget.name.split('_')[1];
+            boxfilter[index] = !boxfilter[parseInt(index)];
+            setSelectedFilter(boxfilter);
+            predefinedValues.forEach((filtre, filterIndex) => {
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/filter-menu.js:56]: filtre renamed to filt**er**

File: src/components/global-view-core.js [hunk 4]
```diff
@@ -247,6 +292,15 @@ const GlobalViewCore = ({ timestampMin, timestampMax, timestampStep }) => {
                         <TableRow>
                             <TableCell size="small">
                                 <FormattedMessage id="timestamp" />
+                                <FilterMenu
+                                    filterHint="filterOnTimestamp"
+                                    handleChange={handletimestampFilterChange}
+                                    currentFilter={statusFilter}
+                                    manual={
+                                        fixedFilter.length > 0 ? false : true
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/global-view-core.js:300]: yes you're right

File: src/components/global-view-core.js [hunk 5]
```diff
@@ -136,21 +152,49 @@ const GlobalViewCore = ({ timestampMin, timestampMax, timestampStep }) => {
         setModalFileOpen(false);
     };
 
-    const handleStatusFilterChange = (event) => {
-        let newFilter = event.currentTarget.value.toUpperCase();
+    const handleStatusFilterChange = (filtres) => {
+        let newFilter = filtres.map((filtre) => filtre.toUpperCase());
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/global-view-core.js:156]: the four occurrences of filtres has been renamed 
and
the four occurrences of filtre has been renamed

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**